doc/source/tutorial/interpolate/1D.rst

```diff
@@ -93,7 +93,8 @@ by the data. `scipy.interpolate` provides two objects of this kind,
 
 A further option for nonparametric interpolation is isotonic regression via
 `IsotonicInterpolator` which constructs a linear interpolant with constant
-regions.
+regions. It is the only interpolator in scipy that enforces a truly
+monotonically increasing solution.
```

Comment:
This statement is incorrect. cf `PchipInterpolator`, `Akima1DInterpolator`

Reply:
I thought this myself, too. But then I had a look at the tutorial, https://docs.scipy.org/doc/scipy/tutorial/interpolate/1D.html#monotone-interpolants, and after x=4.5 both interpolators decrease.
My guess is that they give monotonic interpolation IF the data is monotonic.

Reply:
I guess I see what you mean, but the statement is still incorrect as written.
First of all, this `IsotonicInterpolant` is not an interpolant: it does not pass through the data points. So it's a "solution" to a different problem.

Updating the tutorial is good, but it should be in a different section. Somewhere closer to smoothing splines I guess. It'd be good to cross-ref it with monotone interpolants, I agree, preferably avoiding value judgements and empty semantics. The difference is not that this is a _truly monotonic_ (and the other ones are what, falsely monotonic?). The main difference, AFAICS is whether interpolation is required or not, and whether the result needs to be locally or globally monotonic (that's not a proper term either). 

Pchip et al only require that for each pair of data points, `(x1, y1)` and `(x2, y2)`, `min(y1, y2) <= f(x) <= max(y1, y2)` for  `x1 <= x <= x2`.

Reply:
I see. Please excuse my confusion with terms. In this case, isotonic regression is not an interpolation but curve fitting under the constrained to be always monotonic. Should we rename it into `IsotonicCurveFit`, `IsotonicRegression` or something like that? Under which section in the `interpolate` module should it be placed?

BTW, I could open a PR to make the point with `PchipInterpolator` clearer, i.e. `PchipInterpolator.__call__(x)` will be a monotonic curve iff
- `y` in the fitting call `PchipInterpolator(x, y)` was monotone increasing
- `xnew` in subsequent `PchipInterpolator(xnew)` (`__call__` method) is increasing.

Reply:
Documentation updates are certainly most welcome, please make that doc PR! I'm not sure what is your point about the subsequent call, but let's discuss it in that future PR.

Reply:
W.r.t. this isotonic thing. First and foremost, I now learned something, thank you @lorentzenchr and @WarrenWeckesser. 

Let's start from the beginning. Since it's a proposal for new functionality, it should go through the scipy-dev mailing list. Things to decide on the ML:
- Do we add it at all? IOW, is it common enough, is it used in a wide variety of contexts etc.
- Which part of scipy do we add it? `scipy.interpolate` sounds like a reasonable choice indeed. But now that you mention `curve_fit`, maybe it fits into `scipy.optimize`? Warren's example also used minimize (or is it a technical detail of how to solve the underlying regression problem).
- Naming. If it goes to `interpolate`, it's definitely not `YYYCurveFit`. (It does not sound like curve fitting to me, but I have just only learned about its existence).

If it goes to scipy.interpolate, it's somewhere close to the smoothing splines. While this is sufficiently different, it seems to fail under a general "regression" umbrella (again, I've only learned about it, so take this with salt). So we might make a separate category in the tutorial, move the FITPACK splines into it as a subsection and add this (and, in time, `make_smoothing_spline`, however this is separate from this discussion.)

Reply:
Meanwhile, I would put the pava algorithm into scipy.optimize as it solves an optimization problem. No sure though where the isotonic regression should go.

Reply:
If we remove `IsotonicInterpolator` in the end, these additions to the tutorial on interpolators would also be removed. Let's see...

Reply:
I'll place Warrens example in `isotonic_regression` (the Python wrapper around pava).